jobs/haproxy/templates/haproxy.config.erb

```diff
@@ -23,6 +23,9 @@ defaults
     timeout http-keep-alive <%= p("ha_proxy.keepalive_timeout").to_i  * 1000 %>ms
     timeout http-request    <%= p("ha_proxy.request_timeout").to_i    * 1000 %>ms
     timeout queue           <%= p("ha_proxy.queue_timeout").to_i      * 1000 %>ms
+    <% if p("ha_proxy.init_addr") %>
```

Comment:
The init_addr is independent of the resolver and will deploy successfully with and without the resolver section.  Previously the deploy would fail because with Vault the domain name disappears when bosh restarted the Vault process and HAProxy error's out when it tried to restart.

The resolvers section was needed to make it useful.  I had to add

```
        resolvers:
          - consul: 127.0.0.1
```

so it would query consul in this case.  Without the resolver section, it would use the default behaviour.

I coded this behaviour instead of a "true/false" for my use case because there are many variations that make sense and exposing the raw line makes using this much simpler.